libsolidity/analysis/TypeChecker.cpp

```diff
@@ -753,6 +753,16 @@ bool TypeChecker::visit(InlineAssembly const& _inlineAssembly)
 			}
 			if (var->isConstant())
 			{
+				if (isConstantVariableRecursive(*var))
+				{
+					m_errorReporter.typeError(
+						3558_error,
+						_identifier.location,
+						"Constant variable is circular."
```

Comment:
Would `Constant variable references itself.` be easier to understand?

Reply:
Not sure if it would be accurate, for `a` when the relations are `a -> b`, `b -> b`.

Reply:
Ah right!